corehq/apps/accounting/forms.py

```diff
@@ -1372,26 +1360,22 @@ def _retrieve_feature_rate(self, rate_form):
         return current_rate
 
     def _retrieve_product_rate(self, rate_form):
-        product = SoftwareProduct.objects.get(id=rate_form['product_id'].value())
-        new_rate = rate_form.get_instance(product)
+        new_rate = rate_form.get_instance()
         if rate_form.is_new():
             # a brand new rate
             self.is_update = True
             return new_rate
-        if product.id not in self.current_products_to_rates:
-            # the plan does not have this rate yet, compare any changes to the feature's current latest rate
-            # also mark the form as updated
-            current_rate = product.get_rate(default_instance=False)
-            if current_rate is None:
+        try:
+            current_rate = SoftwareProductRate.objects.get(id=rate_form['rate_id'].value())
+            # note: custom implementation of SoftwareProductRate.__eq__ here...
+            if not current_rate == new_rate:
```

Comment:
Should this be `!=`? Isn't this equivalent to `(not current_rate) == new_rate`, which is probably not what you want?

Reply:
```===``` is being used because ```SoftwareProductRate``` has a custom implementation of ```__eq__```

This expression is safe because ```===``` takes precedence over ```not``` in Python: http://www.informit.com/articles/article.aspx?p=459269&seqNum=11

Agree though that this is not clear and I'll introduce a followup PR with parentheses.

Reply:
Thanks for the link - yeah, parens would be nice for clarity. I've been doing more js than python lately, where `!` has precedence over `==`.